src/ide/editor/language.ts

```diff
@@ -264,7 +271,7 @@ export class Language {
 					let keywords = rule
 						.substring(pos, rule.length - 1)
 						.split(" ");
-					let prefixes = new Map<Uint32Array, number>(); // map prefix to state index
+					const prefixes = new Map<string, number>(); // map prefix to state index
```

Comment:
TypeScript complained in line 283 (`prefixes.has(prefix)`), that `prefix` is of type `string` while this map expects keys of type `Uint32Array`. I'm pretty sure that the key type should simply be `string`…?

Reply:
I agree. At that point the (partial) keywords are represented as strings, as is evident from `kw` and `prefix`.